lib/Parse/ParseDecl.cpp

```diff
@@ -896,12 +948,15 @@ Parser::parseDifferentiableAttribute(SourceLoc atLoc, SourceLoc loc) {
   }
 
   // Parse 'adjoint: <func_name>'.
-  FuncSpec adjointSpec;
-  if (parseFuncSpec("adjoint", adjointSpec))
-    return errorAndSkipToEnd();
+  {
+    SyntaxParsingContext AdjointContext(
+        SyntaxContext, SyntaxKind::DifferentiableAttributeFuncSpecifier);
+    if (parseFuncSpec("adjoint", adjointSpec))
+      return errorAndSkipToEnd();
+  }
 
   // Parse a trailing 'where' clause if any.
-  TrailingWhereClause *whereClause;
+  // TrailingWhereClause *whereClause;
```

Comment:
Why not remove this line?

Reply:
I missed this, thanks! Removed in https://github.com/apple/swift/pull/17424/commits/0b2e25c6f84386045ccf3c3304023f8fa4edcd87.